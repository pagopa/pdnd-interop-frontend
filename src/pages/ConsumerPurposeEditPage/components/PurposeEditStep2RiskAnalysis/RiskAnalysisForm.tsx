import React from 'react'
import { SectionContainer, SectionContainerSkeleton } from '@/components/layout/containers'
import riskAnalysis from '@/static/risk-analysis/pa/v2.0.json'
import { Alert, Box, Stack, Typography } from '@mui/material'
import { FormProvider } from 'react-hook-form'
import { useTranslation } from 'react-i18next'
import useRiskAnalysisForm from '../../hooks/useRiskAnalysisForm'
import type { RiskAnalysis } from '../../types/risk-analysis.types'
import { StepActions } from '@/components/shared/StepActions'
import type { ActiveStepProps } from '@/hooks/useActiveStep'
import type { Purpose } from '@/api/api.generatedTypes'

type RiskAnalysisFormProps = ActiveStepProps & {
  purpose: Purpose
}

export const RiskAnalysisForm: React.FC<RiskAnalysisFormProps> = ({ back, forward, purpose }) => {
  const { t } = useTranslation('purpose', { keyPrefix: 'edit' })

  const { handleSubmit, formMethods, formComponents, isSubmitBtnDisabled } = useRiskAnalysisForm(
    riskAnalysis as RiskAnalysis,
    purpose,
    forward
  )

  return (
    <FormProvider {...formMethods}>
      <Box component="form" noValidate onSubmit={handleSubmit}>
        <SectionContainer>
          <Typography component="h2" variant="h5">
            {t('step2.title')}
          </Typography>
          <Typography color="text.secondary">{t('step2.description')}</Typography>
          <Alert sx={{ mt: 2, mb: -1 }} severity="warning">
            {t('step2.personalInfoAlert')}
          </Alert>
          <Stack spacing={5}>{formComponents}</Stack>
        </SectionContainer>
        <StepActions
          back={{ label: t('backWithoutSaveBtn'), type: 'button', onClick: back }}
          forward={{
            label: t('forwardWithSaveBtn'),
            disabled: isSubmitBtnDisabled,
            type: 'submit',
          }}
        />
      </Box>
    </FormProvider>
  )
}

export const RiskAnalysisFormSkeleton: React.FC = () => {
  return <SectionContainerSkeleton height={600} />
}
