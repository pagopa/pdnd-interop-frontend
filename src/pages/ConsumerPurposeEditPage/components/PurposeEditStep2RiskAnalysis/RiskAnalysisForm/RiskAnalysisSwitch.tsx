import React from 'react'
import { FormLabel, Switch as MUISwitch, Typography, Stack } from '@mui/material'
import type { SwitchProps as MUISwitchProps } from '@mui/material'
import { Controller, useFormContext } from 'react-hook-form'
import { InputWrapper } from '@/components/shared/InputWrapper'
import { mapValidationErrorMessages } from '@/utils/validation.utils'
import { useTranslation } from 'react-i18next'
import type { ControllerProps } from 'react-hook-form/dist/types'
import type { InputOption } from '@/types/common.types'

type RiskAnalysisSwitchProps = Omit<MUISwitchProps, 'checked' | 'onChange'> & {
  label: string
  infoLabel?: string | JSX.Element
  options: Array<InputOption>
  name: string
  rules?: ControllerProps['rules']
  onValueChange?: (value: boolean) => void
}

export const RiskAnalysisSwitch: React.FC<RiskAnalysisSwitchProps> = ({
  label,
  infoLabel,
  options,
  name,
  sx,
  rules,
  onValueChange,
  ...switchProps
}) => {
  const { formState } = useFormContext()
  const { t } = useTranslation()

  const error = formState.errors[name]?.message as string | undefined

  return (
    <InputWrapper error={error} sx={sx} infoLabel={infoLabel}>
      <FormLabel sx={{ color: 'text.primary' }}>
        <Typography component="span" variant="body1">
          {label}
        </Typography>
        <Stack sx={{ mt: 2, mb: 1 }} direction="row" alignItems="center" spacing={0.25}>
          <Controller
            name={name}
            rules={mapValidationErrorMessages(rules, t)}
            render={({ field: { value, ref, onChange, ...fieldProps } }) => (
              <MUISwitch
                {...switchProps}
                {...fieldProps}
                onChange={(e) => {
                  if (onValueChange) onValueChange(e.target.checked)
                  onChange(e.target.checked)
                }}
                checked={value}
                inputRef={ref}
              />
            )}
          />
          {options.length > 0 && (
            <Typography component="span" variant="body2" fontWeight={600}>
              {options[0].label}
            </Typography>
          )}
        </Stack>
      </FormLabel>
    </InputWrapper>
  )
}
