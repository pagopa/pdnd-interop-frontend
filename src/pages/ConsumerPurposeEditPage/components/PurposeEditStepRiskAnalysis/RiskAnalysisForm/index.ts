export * from './RiskAnalysisForm'
export * from './RiskAnalysisVersionMismatchDialog'
