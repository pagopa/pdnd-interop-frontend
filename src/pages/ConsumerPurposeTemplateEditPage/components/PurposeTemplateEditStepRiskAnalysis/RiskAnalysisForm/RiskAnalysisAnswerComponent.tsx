import { RHFSwitch } from '@/components/shared/react-hook-form-inputs'
import { useTranslation } from 'react-i18next'
import AddIcon from '@mui/icons-material/Add'
import { ButtonNaked } from '@pagopa/mui-italia'
import { useDrawerState } from '@/hooks/useDrawerState'
import { AddAnnotationDrawer } from '@/components/shared/AddAnnotationDrawer'

export const RiskAnalysisAnswerComponent: React.FC<{ questionId: string }> = ({ questionId }) => {
  const { t } = useTranslation('purposeTemplate', { keyPrefix: 'edit.step3' })

  const { isOpen, openDrawer, closeDrawer } = useDrawerState()

  const handleClick = () => {
    openDrawer()
  }

  return (
    <>
      <RHFSwitch
        id={questionId}
        label={t('switchLabel')}
        name={`assignToTemplateUsers.${questionId}`}
        disabled={false}
        sx={{ my: 2, ml: 2 }}
      />
      <ButtonNaked
        color="primary"
        type="button"
        sx={{ fontWeight: 700 }}
        startIcon={<AddIcon fontSize="small" />}
        onClick={handleClick}
      >
        {t('addAnnotationBtn')}
      </ButtonNaked>
      <AddAnnotationDrawer isOpen={isOpen} onClose={closeDrawer} onSubmit={() => {}} />
    </>
  )
}
