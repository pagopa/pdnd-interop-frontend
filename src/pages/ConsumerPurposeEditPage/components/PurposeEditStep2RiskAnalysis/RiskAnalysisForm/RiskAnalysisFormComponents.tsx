import React from 'react'
import type { FormConfigQuestion } from '@/api/api.generatedTypes'
import type { Answers, Questions } from '../../../types/risk-analysis-form.types'
import { useFormContext } from 'react-hook-form'
import {
  RHFCheckboxGroup,
  RHFRadioGroup,
  RHFSelect,
  RHFTextField,
} from '@/components/shared/react-hook-form-inputs'
import { RiskAnalysisSwitch } from './components/RiskAnalysisSwitch'
import { useTranslation } from 'react-i18next'
import useCurrentLanguage from '@/hooks/useCurrentLanguage'
import {
  formatRiskAnalysisInputInfoLabel,
  formatRiskAnalysisInputLabel,
  formatRiskAnalysisInputValidationInfoLabel,
  getRiskAnalysisInputOptions,
} from '@/pages/ConsumerPurposeEditPage/utils/risk-analysis-form.utils'
import RiskAnalysisFormSection from './components/RiskAnalysisFormSection'

/**
 * Returns the updated form components.
 *
 * @param questions - the actual updated questions visible to the user
 * @returns Array of components that should be rendered inside the form
 * */
export const RiskAnalysisFormComponents: React.FC<{ questions: Questions }> = ({ questions }) => {
  const { t } = useTranslation('purpose')
  const lang = useCurrentLanguage()
  const answers = useFormContext<Answers>().watch()

  return React.useMemo(() => {
    function buildFormQuestionComponents(question: FormConfigQuestion) {
      const questionComponents: Array<React.ReactNode> = []

      const maxLength = question?.validation?.maxLength

      const inputOptions = getRiskAnalysisInputOptions(question, answers, lang)
      const label = formatRiskAnalysisInputLabel(question, lang, t)
      const infoLabel = formatRiskAnalysisInputInfoLabel(question, lang)

      const validationInfoLabel = formatRiskAnalysisInputValidationInfoLabel(question, t)

      const sx = { mb: 0 }
      const commonProps = {
        key: question.id,
        name: question.id,
        infoLabel: validationInfoLabel,
        sx,
      }

      switch (question.visualType) {
        case 'text':
          questionComponents.push(
            <RiskAnalysisFormSection
              title={label}
              description={infoLabel}
              formFieldName={question.id}
            >
              <RHFTextField
                {...commonProps}
                inputProps={{ maxLength }}
                rules={{ required: true }}
              />
            </RiskAnalysisFormSection>
          )
          break
        case 'select-one':
          questionComponents.push(
            <RiskAnalysisFormSection
              title={label}
              description={infoLabel}
              formFieldName={question.id}
            >
              <RHFSelect
                {...commonProps}
                options={inputOptions}
                emptyLabel={t('edit.step2.emptyLabel')}
                rules={{ required: true }}
              />
            </RiskAnalysisFormSection>
          )
          break
        case 'checkbox':
          questionComponents.push(
            <RiskAnalysisFormSection
              title={label}
              description={infoLabel}
              formFieldName={question.id}
            >
              <RHFCheckboxGroup
                {...commonProps}
                options={inputOptions}
                rules={{
                  validate: (value) =>
                    (typeof value !== 'undefined' && value.length > 0) ||
                    t('edit.step2.multiCheckboxField.validation.mixed.required'),
                }}
              />
            </RiskAnalysisFormSection>
          )
          break
        case 'radio':
          questionComponents.push(
            <RiskAnalysisFormSection
              title={label}
              description={infoLabel}
              formFieldName={question.id}
            >
              <RHFRadioGroup {...commonProps} options={inputOptions} rules={{ required: true }} />
            </RiskAnalysisFormSection>
          )
          break
        case 'switch':
          questionComponents.push(
            <RiskAnalysisFormSection
              title={label}
              description={infoLabel}
              formFieldName={question.id}
            >
              <RiskAnalysisSwitch
                {...commonProps}
                options={inputOptions}
                rules={{
                  validate: (value) =>
                    (typeof value === 'boolean' && value === true) ||
                    t('edit.step2.riskAnalysisSwitch.validation.boolean.isValue'),
                }}
              />
            </RiskAnalysisFormSection>
          )
          break
      }

      return questionComponents
    }

    const formComponents: Array<React.ReactNode> = []
    const questionIds = Object.keys(questions)

    questionIds.forEach((questionId) => {
      const question = questions[questionId]
      formComponents.push(...buildFormQuestionComponents(question))
    })

    return <>{formComponents}</>
  }, [lang, questions, t, answers])
}
