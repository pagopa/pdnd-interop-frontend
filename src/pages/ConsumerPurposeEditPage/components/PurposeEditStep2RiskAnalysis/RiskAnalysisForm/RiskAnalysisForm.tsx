import React from 'react'
import { SectionContainer, SectionContainerSkeleton } from '@/components/layout/containers'
import { Alert, Box, Stack } from '@mui/material'
import { FormProvider, useForm } from 'react-hook-form'
import { useTranslation } from 'react-i18next'
import type { RiskAnalysisFormConfig } from '@/api/api.generatedTypes'
import { StepActions } from '@/components/shared/StepActions'
import SaveIcon from '@mui/icons-material/Save'
import ArrowBackIcon from '@mui/icons-material/ArrowBack'
import type {
  Answers,
  Questions,
} from '@/components/shared/RiskAnalysisFormComponents/types/risk-analysis-form.types'
import {
  getRiskAnalysisDefaultValues,
  getUpdatedQuestions,
  getValidAnswers,
} from '@/components/shared/RiskAnalysisFormComponents/utils/risk-analysis-form.utils'
import { RiskAnalysisFormComponents } from '@/components/shared/RiskAnalysisFormComponents'

type RiskAnalysisFormProps = {
  defaultAnswers: Record<string, string[]>
  riskAnalysis: RiskAnalysisFormConfig
  onSubmit: (answers: Record<string, string[]>) => void
  onCancel: VoidFunction
}

export const RiskAnalysisForm: React.FC<RiskAnalysisFormProps> = ({
  defaultAnswers,
  riskAnalysis,
  onSubmit,
  onCancel,
}) => {
  const { t } = useTranslation('purpose', { keyPrefix: 'edit' })

  const [_, startTransition] = React.useTransition()
  const [defaultValues, __] = React.useState<Answers>(() =>
    getRiskAnalysisDefaultValues(riskAnalysis.questions, defaultAnswers)
  )
  const [questions, setQuestions] = React.useState<Questions>(() =>
    getUpdatedQuestions(defaultValues, riskAnalysis.questions)
  )

  const formMethods = useForm<Answers>({
    defaultValues,
    mode: 'onSubmit',
    reValidateMode: 'onSubmit',
  })

  const { watch } = formMethods

  /**
   * Subscribes to the form values changes
   * and updates the actual visible questions on values change.
   */
  React.useEffect(() => {
    const subscription = watch((answers) => {
      startTransition(() => {
        setQuestions(getUpdatedQuestions(answers as Answers, riskAnalysis.questions))
      })
    })
    return () => subscription.unsubscribe()
  }, [watch, riskAnalysis])

  const handleSubmit = formMethods.handleSubmit((answers) => {
    const currentQuestionsIds = Object.keys(questions)
    const validAnswers = getValidAnswers(currentQuestionsIds, answers)

    onSubmit(validAnswers)
  })

  return (
    <FormProvider {...formMethods}>
      <Box component="form" noValidate onSubmit={handleSubmit}>
        <SectionContainer title={t('step2.title')} description={t('step2.description')}>
          <Alert sx={{ mt: 2, mb: -1 }} severity="warning">
            {t('step2.personalInfoAlert')}
          </Alert>
        </SectionContainer>
        <Stack spacing={2}>
          <RiskAnalysisFormComponents questions={questions} />
        </Stack>
        <StepActions
          back={{
            label: t('backWithoutSaveBtn'),
            type: 'button',
            onClick: onCancel,
            startIcon: <ArrowBackIcon />,
          }}
          forward={{
            label: t('forwardWithSaveBtn'),
            type: 'submit',
            startIcon: <SaveIcon />,
          }}
        />
      </Box>
    </FormProvider>
  )
}

export const RiskAnalysisFormSkeleton: React.FC = () => {
  return <SectionContainerSkeleton height={600} />
}
