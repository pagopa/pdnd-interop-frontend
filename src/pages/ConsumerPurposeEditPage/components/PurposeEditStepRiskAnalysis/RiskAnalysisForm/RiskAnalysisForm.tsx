import React from 'react'
import { SectionContainer, SectionContainerSkeleton } from '@/components/layout/containers'
import { Alert, Box, Stack } from '@mui/material'
import { FormProvider } from 'react-hook-form'
import { useTranslation } from 'react-i18next'
import type { RiskAnalysisFormConfig } from '@/api/api.generatedTypes'
import { StepActions } from '@/components/shared/StepActions'
import SaveIcon from '@mui/icons-material/Save'
import ArrowBackIcon from '@mui/icons-material/ArrowBack'
import { RiskAnalysisFormComponents } from '@/components/shared/RiskAnalysisFormComponents'
import { useRiskAnalysisForm } from '@/hooks/useRiskAnalysisForm'

type RiskAnalysisFormProps = {
  defaultAnswers: Record<string, string[]>
  riskAnalysis: RiskAnalysisFormConfig
  onSubmit: (answers: Record<string, string[]>) => void
  onCancel: VoidFunction
}

export const RiskAnalysisForm: React.FC<RiskAnalysisFormProps> = ({
  defaultAnswers,
  riskAnalysis,
  onSubmit,
  onCancel,
}) => {
  const { t } = useTranslation('purpose', { keyPrefix: 'edit' })

  const riskAnalysisForm = useRiskAnalysisForm({
    riskAnalysisConfig: riskAnalysis,
    defaultAnswers: defaultAnswers,
  })

  const handleSubmit = riskAnalysisForm.handleSubmit(({ validAnswers }) => onSubmit(validAnswers))

  return (
    <FormProvider {...riskAnalysisForm}>
      <Box component="form" noValidate onSubmit={handleSubmit}>
        <SectionContainer
          title={t('stepRiskAnalysis.title')}
          description={t('stepRiskAnalysis.description')}
        >
          <Alert sx={{ mt: 2, mb: -1 }} severity="warning">
            {t('stepRiskAnalysis.personalInfoAlert')}
          </Alert>
        </SectionContainer>
        <Stack spacing={2}>
          <RiskAnalysisFormComponents questions={riskAnalysisForm.questions} />
        </Stack>
        <StepActions
          back={{
            label: t('backWithoutSaveBtn'),
            type: 'button',
            onClick: onCancel,
            startIcon: <ArrowBackIcon />,
          }}
          forward={{
            label: t('endWithSaveBtn'),
            type: 'submit',
            startIcon: <SaveIcon />,
          }}
        />
      </Box>
    </FormProvider>
  )
}

export const RiskAnalysisFormSkeleton: React.FC = () => {
  return <SectionContainerSkeleton height={600} />
}
