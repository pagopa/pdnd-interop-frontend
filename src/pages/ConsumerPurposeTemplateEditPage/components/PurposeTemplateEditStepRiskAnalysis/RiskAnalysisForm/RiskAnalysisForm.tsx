import React from 'react'
import { SectionContainer, SectionContainerSkeleton } from '@/components/layout/containers'
import { Alert, Box, Stack } from '@mui/material'
import { FormProvider } from 'react-hook-form'
import { useTranslation } from 'react-i18next'
import type { RiskAnalysisFormConfig } from '@/api/api.generatedTypes'
import { StepActions } from '@/components/shared/StepActions'
import SaveIcon from '@mui/icons-material/Save'
import ArrowBackIcon from '@mui/icons-material/ArrowBack'
import { RiskAnalysisFormComponents } from '@/components/shared/RiskAnalysisFormComponents'
import { useRiskAnalysisForm } from '@/hooks/useRiskAnalysisForm'

type RiskAnalysisFormProps = {
  defaultAnswers: Record<string, string[]>
  riskAnalysis: RiskAnalysisFormConfig
  onSubmit: (answers: Record<string, string[]>) => void
  onCancel: VoidFunction
}

export const RiskAnalysisForm: React.FC<RiskAnalysisFormProps> = ({
  defaultAnswers,
  riskAnalysis,
  onSubmit,
  onCancel,
}) => {
  const { t } = useTranslation('purposeTemplate', { keyPrefix: 'edit' })

  const riskAnalysisForm = useRiskAnalysisForm({
    riskAnalysisConfig: riskAnalysis,
    defaultAnswers: defaultAnswers,
  })

  const handleSubmit = riskAnalysisForm.handleSubmit(({ validAnswers }) => onSubmit(validAnswers))

  return (
    <FormProvider {...riskAnalysisForm}>
      <Box component="form" noValidate onSubmit={handleSubmit}>
        <SectionContainer
          title={t('step3.detailsTitle')}
          description={t('step3.detailsDescription')}
        >
          <Alert sx={{ mt: 2, mb: -1 }} severity="warning">
            {t('step3.alert')}
          </Alert>
        </SectionContainer>
        <Stack spacing={2}>
          <RiskAnalysisFormComponents
            questions={riskAnalysisForm.questions}
            isFromPurposeTemplate={true}
          />
        </Stack>
        <StepActions
          back={{
            label: t('backWithoutSaveBtn'),
            type: 'button',
            onClick: onCancel,
            startIcon: <ArrowBackIcon />,
          }}
          forward={{
            label: t('forwardWithSaveBtn'),
            type: 'submit',
            startIcon: <SaveIcon />,
          }}
        />
      </Box>
    </FormProvider>
  )
}

export const RiskAnalysisFormSkeleton: React.FC = () => {
  return <SectionContainerSkeleton height={600} />
}
