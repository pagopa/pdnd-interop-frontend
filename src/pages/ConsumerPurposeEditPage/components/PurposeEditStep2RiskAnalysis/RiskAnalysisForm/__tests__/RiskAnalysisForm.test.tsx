import React from 'react'
import { fireEvent, render, waitFor } from '@testing-library/react'
import { RiskAnalysisForm, RiskAnalysisFormSkeleton } from '../RiskAnalysisForm'
import { createMockRiskAnalysisFormConfig } from '__mocks__/data/purpose.mocks'
import { vi } from 'vitest'
import userEvent from '@testing-library/user-event'

describe('RiskAnalysisForm', () => {
  it('should match the snapshot', () => {
    const { baseElement } = render(
      <RiskAnalysisForm
        defaultAnswers={{}}
        riskAnalysis={createMockRiskAnalysisFormConfig()}
        onSubmit={vi.fn()}
        onCancel={vi.fn()}
      />
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('should update the questions accordingly', async () => {
    const screen = render(
      <RiskAnalysisForm
        defaultAnswers={{}}
        riskAnalysis={createMockRiskAnalysisFormConfig()}
        onSubmit={vi.fn()}
        onCancel={vi.fn()}
      />
    )

    expect(screen.queryByRole('textbox', { name: '' })).not.toBeInTheDocument()

    fireEvent.click(
      screen.getByRole('radio', {
        name: 'Other',
      })
    )

    expect(screen.getByRole('textbox', { name: '' })).toBeInTheDocument()
  })

  it('should submit the form with the correct values', async () => {
    const onSubmit = vi.fn()
    const screen = render(
      <RiskAnalysisForm
        defaultAnswers={{}}
        riskAnalysis={createMockRiskAnalysisFormConfig()}
        onSubmit={onSubmit}
        onCancel={vi.fn()}
      />
    )

    const user = userEvent.setup()

    fireEvent.click(
      screen.getByRole('radio', {
        name: 'Other',
      })
    )

    await user.type(screen.getByRole('textbox', { name: '' }), 'Some text')

    fireEvent.click(screen.getByRole('button', { name: 'forwardWithSaveBtn' }))

    await waitFor(() => {
      expect(onSubmit).toHaveBeenCalledWith({
        purpose: ['OTHER'],
        institutionalPurpose: ['Some text'],
      })
    })
  })
})

describe('RiskAnalysisFormSkeleton', () => {
  it('should match snapshot', () => {
    const { baseElement } = render(<RiskAnalysisFormSkeleton />)
    expect(baseElement).toMatchSnapshot()
  })
})
