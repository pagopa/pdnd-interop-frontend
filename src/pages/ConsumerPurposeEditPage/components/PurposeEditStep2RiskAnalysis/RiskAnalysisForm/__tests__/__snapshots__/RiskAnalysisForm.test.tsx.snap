// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RiskAnalysisForm > should match the snapshot 1`] = `
<body>
  <div>
    <form
      class="MuiBox-root css-0"
      novalidate=""
    >
      <section
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-u52cu0-MuiPaper-root"
      >
        <div
          class="MuiStack-root css-jfdv4h-MuiStack-root"
        >
          <div
            class="MuiStack-root css-3y1137-MuiStack-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              step2.title
            </h2>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            step2.description
          </p>
        </div>
        <div
          class="MuiBox-root css-1yuhvjn"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-1w4fysv-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="ReportProblemOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
            >
              step2.personalInfoAlert
            </div>
          </div>
        </div>
      </section>
      <div
        class="MuiStack-root css-1sazv7p-MuiStack-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-u52cu0-MuiPaper-root"
        >
          <div
            class="MuiStack-root css-jfdv4h-MuiStack-root"
          >
            <div
              class="MuiStack-root css-3y1137-MuiStack-root"
            />
          </div>
          <div
            class="MuiBox-root css-zv7ju9"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <div
                class="MuiStack-root css-givlym-MuiStack-root"
              >
                <label
                  class="MuiFormLabel-root MuiFormLabel-colorPrimary css-j6w7dp-MuiFormLabel-root"
                  id="purpose-label"
                >
                  Question 1 (edit.step2.validation.required)
                </label>
              </div>
              <div
                aria-labelledby="purpose-label"
                class="MuiFormGroup-root css-1w0y3ky-MuiFormGroup-root"
                id="purpose"
                role="radiogroup"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-5g7fen-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      name="purpose"
                      type="radio"
                      value="INSTITUTIONAL"
                    />
                    <span
                      class="css-hyxlzm"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                        data-testid="RadioButtonUncheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                        data-testid="RadioButtonCheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    option 1
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-5g7fen-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      name="purpose"
                      type="radio"
                      value="OTHER"
                    />
                    <span
                      class="css-hyxlzm"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                        data-testid="RadioButtonUncheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                        data-testid="RadioButtonCheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    Other
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiStack-root css-1itaudw-MuiStack-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowBackIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </span>
          backWithoutSaveBtn
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SaveIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
              />
            </svg>
          </span>
          forwardWithSaveBtn
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</body>
`;

exports[`RiskAnalysisFormSkeleton > should match snapshot 1`] = `
<body>
  <div>
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1u7iav8-MuiSkeleton-root"
      style="height: 600px;"
    />
  </div>
</body>
`;
